meter/hoymiles_wifi.go

```diff
@@ -0,0 +1,114 @@
+// MIT License
+//
+// Copyright (c) 2024 Björn Lundström - https://github.com/BLun78
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package meter
+
+import (
+	"net"
+	"os"
+	"syscall"
+	"time"
+
+	"github.com/BLun78/hoymiles_wifi"
+	"github.com/BLun78/hoymiles_wifi/hoymiles/models"
+	"github.com/evcc-io/evcc/api"
+	"github.com/evcc-io/evcc/util"
+)
+
+func init() {
+	registry.Add("hoymiles-wifi", NewHoymilesWifiMeterFromConfig)
+}
+
+type HoymilesWifi struct {
+	client           *hoymiles_wifi.ClientData
+	log              *util.Logger
+	cc               struct{ Host string }
+	lastValue        float64
+	lastValueUpdated time.Time
+}
+
+func NewHoymilesWifiMeterFromConfig(other map[string]interface{}) (api.Meter, error) {
+	var cc struct {
+		Host string
+	}
+
+	if err := util.DecodeOther(other, &cc); err != nil {
+		return nil, err
+	}
+
+	log := util.NewLogger("hoymiles-wifi")
+	log.TRACE.Printf("Start HoymilesWifi setup: %s", cc.Host)
+
+	client := hoymiles_wifi.NewClientDefault(cc.Host)
+	client.ConnectionTimeout = 5 * time.Second
+
+	return &HoymilesWifi{
+		client:    client,
+		log:       log,
+		cc:        cc,
+		lastValue: 0,
+	}, nil
+}
+
+// CurrentPower implements the api.Meter interface
+func (hmWifi *HoymilesWifi) CurrentPower() (float64, error) {
+	var value float64
+	request := &models.RealDataNewReqDTO{}
+	// int32 would not be Year 2038 safe
+	// See https://en.wikipedia.org/wiki/Year_2038_problem
+	// Not 100% sure if the models are self-defined or provided by hoymiles
+	request.Time = int32(time.Now().Unix())
```

Comment:
Das kann in. Die Initialisierung des Requests mit rein. Eigentlich scheint das aber in die Lib zu gehören?